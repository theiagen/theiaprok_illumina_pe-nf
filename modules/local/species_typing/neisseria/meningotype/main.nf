process MENINGOTYPE {
    tag "$meta.id"
    label "process_low"

    container "us-docker.pkg.dev/general-theiagen/biocontainers/meningotype:0.8.5--pyhdfd78af_0"

    input:
    tuple val(meta), path(assembly)

    output:
    tuple val(meta), path("*.tsv")               , emit: meningotype_report
    tuple val(meta), path("MENINGOTYPE_SEROTYPE"), emit: meningotype_serogroup
    tuple val(meta), path("MENINGOTYPE_PORA")    , emit: meningotype_pora
    tuple val(meta), path("MENINGOTYPE_FETA")    , emit: meningotype_feta
    tuple val(meta), path("MENINGOTYPE_PORB")    , emit: meningotype_porb
    tuple val(meta), path("MENINGOTYPE_FHBP")    , emit: meningotype_fhbp
    tuple val(meta), path("MENINGOTYPE_NHBA")    , emit: meningotype_nhba
    tuple val(meta), path("MENINGOTYPE_NADA")    , emit: meningotype_nada
    tuple val(meta), path("MENINGOTYPE_BAST")    , emit: meningotype_bast
    path "versions.yml"                          , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def finetype = params.meningotype_finetype ? '--finetype' : ''
    def porB = params.meningotype_porB ? '--porB' : ''
    def bast = params.meningotype_bast ? '--bast' : ''
    def mlst = params.meningotype_mlst ? '--mlst' : ''
    def all_typing = params.meningotype_all ? '--all' : ''
    // Default behavior matches WDL (finetype, porB, bast enabled)
    def default_finetype = finetype ?: '--finetype'
    def default_porB = porB ?: '--porB'
    def default_bast = bast ?: '--bast'
    
    """
    # Run meningotype on input assembly
    # Parameters:
    # --finetype      perform porA and fetA fine typing (default=off)
    # --porB          perform porB sequence typing (NEIS2020) (default=off)
    # --bast          perform Bexsero antigen sequence typing (BAST) (default=off)
    # --mlst          perform MLST (default=off)
    # --all           perform MLST, porA, fetA, porB, BAST typing (default=off)
    meningotype \\
        ${all_typing} \\
        ${default_finetype} \\
        ${default_porB} \\
        ${default_bast} \\
        ${mlst} \\
        --cpus ${task.cpus} \\
        ${assembly} \\
        ${args} \\
        > ${prefix}.tsv


    tail -1 ${prefix}.tsv | awk '{print \$2}' | tee MENINGOTYPE_SEROTYPE
    tail -1 ${prefix}.tsv | awk '{print \$5}' | tee MENINGOTYPE_PORA
    tail -1 ${prefix}.tsv | awk '{print \$6}' | tee MENINGOTYPE_FETA
    tail -1 ${prefix}.tsv | awk '{print \$7}' | tee MENINGOTYPE_PORB
    tail -1 ${prefix}.tsv | awk '{print \$8}' | tee MENINGOTYPE_FHBP
    tail -1 ${prefix}.tsv | awk '{print \$9}' | tee MENINGOTYPE_NHBA
    tail -1 ${prefix}.tsv | awk '{print \$10}' | tee MENINGOTYPE_NADA
    tail -1 ${prefix}.tsv | awk '{print \$11}' | tee MENINGOTYPE_BAST
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        meningotype: \$(meningotype --version 2>&1 | sed 's/^.*meningotype v//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.tsv
    touch MENINGOTYPE_SEROTYPE
    touch MENINGOTYPE_PORA
    touch MENINGOTYPE_FETA
    touch MENINGOTYPE_PORB
    touch MENINGOTYPE_FHBP
    touch MENINGOTYPE_NHBA
    touch MENINGOTYPE_NADA
    touch MENINGOTYPE_BAST

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        meningotype: \$(meningotype --version 2>&1 | sed 's/^.*meningotype v//' || echo "0.8.5")
    END_VERSIONS
    """
}