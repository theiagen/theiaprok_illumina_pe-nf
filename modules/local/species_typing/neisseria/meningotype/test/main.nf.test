nextflow_process {
    name "Test Process MENINGOTYPE "
    script "../main.nf"
    process "MENINGOTYPE"

    test("MENINGOTYPE  - test") {
        setup {
            run("GUNZIP") {
                script "../../../../../nf-core/gunzip/main.nf"
                process {
                    """
                    input[0] = [
                        [id:'test_sample'],
                        file('https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/019/048/825/GCF_019048825.1_ASM1904882v1/GCF_019048825.1_ASM1904882v1_genomic.fna.gz', checkIfExists: true)
                    ]
                    """
                }
            }
        }
        when {
            process {
                """
                input[0] = GUNZIP.out.gunzip
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.meningotype_report,
                    process.out.meningotype_serogroup,
                    process.out.meningotype_pora,
                    process.out.meningotype_feta,
                    process.out.meningotype_porb,
                    process.out.meningotype_fhbp,
                    process.out.meningotype_nhba,
                    process.out.meningotype_nada,
                    process.out.meningotype_bast,
                    process.out.versions
                ).match() }
            )
        }
    }

    test("MENINGOTYPE  - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test_sample' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/escherichia_coli/genome/genome.fa.gz', checkIfExists: true)
                ])
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}