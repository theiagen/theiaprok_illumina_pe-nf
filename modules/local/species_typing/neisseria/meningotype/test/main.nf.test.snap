{
    "MENINGOTYPE  - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_SEROTYPE:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_PORA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_FETA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_PORB:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_FHBP:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_NHBA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_NADA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_BAST:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    "versions.yml:md5,91abd476dd11f1259a8ef6f8be2fd7a4"
                ],
                "meningotype_bast": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_BAST:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_feta": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_FETA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_fhbp": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_FHBP:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_nada": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_NADA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_nhba": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_NHBA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_pora": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_PORA:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_porb": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_PORB:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_report": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "meningotype_serogroup": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "MENINGOTYPE_SEROTYPE:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,91abd476dd11f1259a8ef6f8be2fd7a4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.5"
        },
        "timestamp": "2025-06-24T20:16:20.840625"
    },
    "MENINGOTYPE  - test": {
        "content": [
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "test_sample.tsv:md5,d9215741782af688916e5408e53a1148"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_SEROTYPE:md5,b39bfc0e26a30024c76e4dcb8a1eae87"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_PORA:md5,5e5d2781cf382826446ff70f0f5e9de6"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_FETA:md5,693a0f9f92783c3ef2979d37a20422e3"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_PORB:md5,6717dc6c302b21d7f8569c28391a67d6"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_FHBP:md5,31d30eea8d0968d6458e0ad0027c9f80"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_NHBA:md5,dbbf8220893d497d403bb9cdf49db7a4"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_NADA:md5,9cd599a3523898e6a12e13ec787da50a"
                ]
            ],
            [
                [
                    {
                        "id": "test_sample"
                    },
                    "MENINGOTYPE_BAST:md5,0a1f21a3417389e0c0a13392c79a7a89"
                ]
            ],
            [
                "versions.yml:md5,91abd476dd11f1259a8ef6f8be2fd7a4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.5"
        },
        "timestamp": "2025-06-24T20:16:16.825177"
    }
}